### PR TITLE
sim_vehicle.py: Added regex based splits

### PR DESCRIPTION
**Pull request by arshPratap**
Added a regex based split for properly parsing the arguments for sending to mavproxy, as suggested [here](https://github.com/arshPratap/ardupilot/blob/master/Tools/autotest/sim_vehicle.py#L718)

### CODE REVIEW

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -715,15 +715,8 @@ def start_mavproxy(opts, stuff):
...
-        # It would be great if this could be done with regex
-        mavargs = opts.mavproxy_args.split(" ")
-        # Find the arguments with '=' in them and split them up
-        for i, x in enumerate(mavargs):
-            if '=' in x:
-                mavargs[i] = x.split('=')[0]
-                mavargs.insert(i+1, x.split('=')[1])
-        # Use this flag to tell if parsing character inbetween a pair
-        # of quotation marks
+        #Split the arguments on space and "=" signs using regex
```
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py:718]:
```suggestion
        # Split the arguments on space and "=" signs using regex
```

File: Tools/autotest/sim_vehicle.py [hunk 2]
```diff
@@ -715,15 +715,8 @@ def start_mavproxy(opts, stuff):
...
+        mavargs = re.split("[\s=]",opts.mavproxy_args)
```
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py:719]:
```suggestion
        mavargs = re.split("[\s=]", opts.mavproxy_args) 
```

**Comment by peterbarker** [comment]
This looks good.  How have you tested it?

**Comment by amilcarlucas** [comment]
I have a mavproxy PR that fixes some command line parsing, you might want to try that.

File: Tools/autotest/sim_vehicle.py (hunk 1 above)
Comment (arshPratap) [reply Tools/autotest/sim_vehicle.py]: Thank you Sir for suggesting the changes.

**Comment by arshPratap** [comment]
@amilcarlucas 
Thank you Sir , I would definitely try it

**Comment by arshPratap** [comment]
@peterbarker Sir, I have made the changes. Would be great if you could take a look.Thanks!

**Review by peterbarker — CHANGES_REQUESTED** [review]
Again, how have you tested this?

The *first* test I tried here fell over:

```
pbarker@bluebottle:~/rc/ardupilot((HEAD detached at arshPratap/Simulation))$ ./Tools/autotest/sim_vehicle.py -v ArduCopter -m --out=localhost:7890
SIM_VEHICLE: Start
SIM_VEHICLE: Killing tasks
SIM_VEHICLE: Starting up at SITL location
Traceback (most recent call last):
  File "./Tools/autotest/sim_vehicle.py", line 1270, in <module>
    do_build(vehicle_dir, cmd_opts, frame_infos)
TypeError: do_build() takes 2 positional arguments but 3 were given
SIM_VEHICLE: Killing tasks
pbarker@bluebottle:~/rc/ardupilot((HEAD detached at arshPratap/Simulation))$ 
```
(that passes on master)

**Comment by peterbarker** [comment]
Closing this one now.  Reopen/create fresh if you still wanted to pursue.

**Pull request closed without merging**